backend/app/users/schemas.py

```diff
@@ -0,0 +1,43 @@
+from datetime import datetime
+from typing import Optional
+
+from pydantic import BaseModel, EmailStr, Field
+
+
+class UserCreate(BaseModel):
+    """
+    Class representing the data required to create a new user.
+    """
+
+    email: str = Field(..., description="The email address of the new user.")
+    password: str = Field(..., description="The password of the new user.")
+    role: Optional[str] = Field(
+        None, description="The role of the new user, if applicable."
+    )
+
+
+class UserOut(BaseModel):
```

Comment:
What about `NewUser` as the name of this class? `UserOut` doesn't seem informative

Reply:
Or maybe just `User`?

Reply:
So the logic here - and I was basically following directly from a tutorial but it made sense to me - was that any time we output users from the DB we should never have their password shown. So basically any `get`/`post` operations we do should follow this schema. So `UserOut` represents showing a User's info publicly.

I do use `UserOut` for other calls -> on creation and deletion too. It basically serves as a nice template for "Hey I did a thing with this user (no password shown) and this was what it was"